src/pages/home/HeaderView.js

```diff
@@ -200,103 +213,107 @@ function HeaderView(props) {
+                                                numberOfLines={1}
+                                            >
+                                                {subtitle}
+                                            </Text>
+                                        )}
                                     </View>
-                                )}
-                            </PressableWithoutFeedback>
-                            <View style={[styles.reportOptions, styles.flexRow, styles.alignItemsCenter]}>
-                                {isTaskReport && !isSmallScreenWidth && ReportUtils.isOpenTaskReport(props.report) && <TaskHeaderActionButton report={props.report} />}
-                                {shouldShowThreeDotsButton && (
-                                    <ThreeDotsMenu
-                                        anchorPosition={styles.threeDotsPopoverOffset(windowWidth)}
-                                        menuItems={threeDotMenuItems}
-                                        shouldSetModalVisibility={false}
-                                    />
-                                )}
+                                    {brickRoadIndicator === CONST.BRICK_ROAD_INDICATOR_STATUS.ERROR && (
+                                        <View style={[styles.alignItemsCenter, styles.justifyContentCenter]}>
+                                            <Icon
+                                                src={Expensicons.DotIndicator}
+                                                fill={theme.danger}
+                                            />
+                                        </View>
+                                    )}
+                                </PressableWithoutFeedback>
+                                <View style={[styles.reportOptions, styles.flexRow, styles.alignItemsCenter]}>
+                                    {isTaskReport && !isSmallScreenWidth && ReportUtils.isOpenTaskReport(props.report) && <TaskHeaderActionButton report={props.report} />}
+                                    {canJoin && !isSmallScreenWidth && joinButton}
+                                    {shouldShowThreeDotsButton && (
+                                        <ThreeDotsMenu
+                                            anchorPosition={styles.threeDotsPopoverOffset(windowWidth)}
+                                            menuItems={threeDotMenuItems}
+                                            shouldSetModalVisibility={false}
+                                        />
+                                    )}
+                                </View>
                             </View>
-                        </View>
-                    </>
-                )}
+                        </>
+                    )}
+                </View>
             </View>
+            {canJoin && isSmallScreenWidth && <View style={[styles.ph5, styles.pb2]}>{joinButton}</View>}
         </View>
```

Comment:
This looks like so many changes, but it just changing

from
```
<View style={[styles.appContentHeader, shouldShowBorderBottom && styles.borderBottom]}
            dataSet={{dragArea: true}}>
     ... (unchanged)
</View>
```

to

```
<View style={[shouldShowBorderBottom && styles.borderBottom]}  dataSet={{dragArea: true}}>
    <View style={[styles.appContentHeader]}>
        ... (unchanged)
    </View>
    {join button for small screen)
</View>
```

Reply:
This seems to better approach than task button which is existing now, since for small screen we render the task button from ReportScreen and non-small screen we render the task button from HeaderView.
I think in future we should have standard component to render these kind of buttons and their logic.